docs/customizing.rst

```diff
@@ -0,0 +1,198 @@
+
+.. code-block:: python
+
+    [
+        {
+        "sortBy": "bestmatch",
+        "text": "Best match"
+        },
+        {
+        "sortBy": "newest",
+        "text": "Newest"
+        }
+    ]
+
+This helper needs to be provided with the ID of the endpoint registered in your
+Invenio-Records-REST configuration.
+https://github.com/inveniosoftware/invenio-records-rest/blob/027794e6ae8dbe2448c59a64b30cdd3c4abb7eb7/invenio_records_rest/config.py#L26
+
+.. code-block:: python
+
+    SearchAppInvenioRestConfigHelper.generate(
+                {'endpoint_id': 'recid'})
+
+It will then generate a react-searchkit compatible configuration, like the
+one mentioned above.
+You can provide this configuration to the jinja template via the
+search_app_helpers macro.
+You can also make adjustments to the default values
+(e.g. page size) by providing your own when initialising the helper object.
+
+.. code-block:: python
+
+    SearchAppInvenioRestConfigHelper({'default_size':10}).generate(
+                {'endpoint_id': 'recid'})
+
+If a part of the automatically generated configuration doesn't suit your case
+you can completely overwite it by supplementing the .generate function with the
+correct one.
+This helper is provided to cover most of the basic cases, but if you feel
+that you need to replace many parts of the generated configuration object
+you can also remove it and provide directly the configuration object.
+
+.. code-block:: python
+
+    # For instance if you want to disable the facets in the UI despite
+    # the aggregations being available in the backend.
+    SearchAppInvenioRestConfigHelper({'default_size':10}).generate(
+                {'endpoint_id': 'recid'}, {'aggs':[]})
+
+Then just add it to your jinja template rendering the search page:
+
+.. code-block:: html+jinja
+
+    <div data-invenio-search-config='{{
+    search_app_helpers.invenio_records_rest.generate(
+        dict(
+        endpoint_id="recid",
+        app_id="search"
+        )
+    ) | tojson(indent=2) }}'></div>
+
+
+
+Configuring the components of your Search Application
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To render a search page you need 3 elements, a jinja template containing a div
+element with the attribute 'data-invenio-search-config' populated with a
+react-searchkit configuration, while also importing an initialised SearchApp
+from invenio-search-ui.
+
+To do the latter you can create a `index.js` file under the `semamantic-ui/js`
+folder and use the `createSearchAppInit` from invenio-search-ui to initialise a
+SearchApp.
+You can point to this additional file by including it to the `entry` object in
+`webpack.py`.
+
+.. code-block:: python
+
+    entry={
+        "invenio_search_ui_app": "./js/invenio_search_ui/app.js",
+    },
+
+Finally just include the created bundle in your jinja template:
+
+.. code-block:: html+jinja
+
+    {%- block javascript %}
+    {{ super() }}
+    {{ webpack['invenio_search_ui_app.js'] }}
```

Comment:
major: This order has to be changed due to how `react-overridable` works. We had a big problem in the cookie-cutter, see [here](https://github.com/inveniosoftware/cookiecutter-invenio-instance/pull/253/files#diff-4842c5a124984dcd66aea9eaaa191570d428ca6e7898619498056573ae2f53c7R6). Basically, the overrides are loaded early (instead of lazy) so if super is called first, you *cannot override anything. See [here](https://github.com/inveniosoftware/invenio-search-ui/blob/master/invenio_search_ui/assets/semantic-ui/js/invenio_search_ui/components/SearchApp.js#L62).
```suggestion
    {{ webpack['invenio_search_ui_app.js'] }}
    {{ super() }}
```